Configure.ps1

```diff
@@ -82,74 +81,75 @@ of the visual studio solution (App.Config and Web.Config) so that they are consi
-    $daemonservicePrincipal = New-AzureRmADServicePrincipal -ApplicationId $daemonApplication.ApplicationId
+    $certBase64Value = [System.Convert]::ToBase64String($certificate.GetRawCertData())
+    $certBase64Thumbprint = [System.Convert]::ToBase64String($certificate.GetCertHash())
+    
+    # Add a Azure Key Credentials from the certificate for the application
+    $applicationKeyCredentials = New-AzureADApplicationKeyCredential -ObjectId $daemonApplication.ObjectId -CustomKeyIdentifier $certificateName -Type AsymmetricX509Cert -Usage Verify -Value $certBase64Value  -StartDate $certificate.NotBefore -EndDate $certificate.NotAfter
 
     # Update the config file in the daemon application
     $configFile = $pwd.Path + "\TodoListDaemonWithCert\App.Config"
     Write-Host "Updating the configuration file for the client ($configFile)"
-    UpdateClientConfigFile -configFilePath $configFile -tenantId $tenantId -clientId $daemonApplication.ApplicationId -appUri $serviceAppIdIRI -baseAddress $serviceHomePage -certificateName $certificateName
+    UpdateClientConfigFile -configFilePath $configFile -tenantId $tenantId -clientId $daemonApplication.AppId -appUri $serviceAppIdURI -baseAddress $serviceHomePage -certificateName $certificateName
 
     # Update tehe config file in the service application
     $configFile = $pwd.Path + "\TodoListService\Web.Config"
     Write-Host "Updating the client sample ($configFile)"
-    UpdateServiceConfigFile -configFilePath $configFile -tenantId $tenantId -audience $serviceAppIdIRI
+    UpdateServiceConfigFile -configFilePath $configFile -tenantId $tenantId -audience $serviceAppIdURI
 
     # prepare the clean-up
-    $applicationsInformation = @{serviceApp=$serviceApplication; serviceAppServicePrincipal=$serviceservicePrincipal; clientApp=$daemonApplication; clientAppServicePrincipal=$daemonservicePrincipal;}
+    $applicationsInformation = @{serviceApp=$serviceApplication; serviceAppServicePrincipal=$serviceservicePrincipal; clientApp=$daemonApplication; clientAppServicePrincipal=$daemonservicePrincipal; certificateThumbPrint=$certBase64Thumbprint}
     return $applicationsInformation
    }
 }
 
 
-#Clean-Ups the applications and serice principal in Azure AD.
-Function CleanUp($applicationsInformation)
+# Remove the AAD applications and service principals registered for the sample.
+Function Cleanup($applicationsInformation)
 {
-    remove-azurermadserviceprincipal -objectid $applicationsInformation.serviceAppServicePrincipal.id -force
-    remove-azurermadapplication -objectid $applicationsInformation.serviceApp.objectid -force
-    remove-azurermadserviceprincipal -objectid $applicationsInformation.clientAppServicePrincipal.id -force
-    remove-azurermadapplication -objectid $applicationsInformation.clientApp.objectid -force
+    remove-azurermadserviceprincipal -objectid $applicationsInformation.serviceAppServicePrincipal.id
```

Comment:
Same goes for variable names too. The PowerShell ISE or VS Code should autocomplete to the correct case them when pressing [Tab].